SteakBot.Core/Modules/MemeModule.cs

```diff
@@ -20,20 +20,50 @@ public MemeModule(MemeService memeService)
         public async Task AddMeme(string name, string value, string description)
         {
             var replyMessage = string.Empty;
-            EmbedBuilder embed = null;
+            EmbedBuilder embed = new EmbedBuilder();
 
-            var type = value.ToLower().Contains("http") ? MemeResultType.Image : MemeResultType.Text;
+            var type = _memeService.GetMemeType(value);
 
             var result = _memeService.AddCommand(new MemeCommand(type, name, value, description));
 
-            if (result)
+            await SendReply(result, type, value, "Add failed! ;(");
+        }
+
+        [Command("editMeme")]
+        public async Task EditMeme(string oldName, string newName, string newValue, string newDescription)
+        {
+            var replyMessage = string.Empty;
+            EmbedBuilder embed = new EmbedBuilder();
+
+            var type = _memeService.GetMemeType(newValue);
+
+            var result = _memeService.EditCommand(oldName, new MemeCommand(type, newName, newValue, newDescription));
+
+            await SendReply(result, type, newValue, "Edit failed! ;(");
+        }
+
+        [Command("removeMeme")]
+        public async Task RemoveMeme(string name)
+        {
+            var result = _memeService.RemoveCommand(name);
+
+            var replyMessage = result ? "Removed" : "Remove failed! ;(";
+
+            await ReplyAsync(replyMessage);
+        }
+
+        #region Private methods
+
+        private Task SendReply(bool commandResult, MemeResultType memeType, string value, string failureMessage)
```

Comment:
Minor nit: the name here sounds like the library functions.

Reply:
Renamed.